config/jobs/image-pushing/k8s-staging-csi-gen.sh

```diff
@@ -0,0 +1,71 @@
+#! /bin/bash -e
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
```

Comment:
In our infra-related scripts at [`k/k8s.io`](https://github.com/kubernetes/k8s.io) we are trying to be consistent with [Google's shell styleguide ](https://google.github.io/styleguide/shellguide.html) so some of my comments will be related to this styleguide, but I'm not gonna block the PR because of this and leave you the decission if you would like to consider the suggestions.

As setting flags directly in shebangs can be broken when calling te script like:
```bash
bash script_name.sh
```
I would suggest
```suggestion
#!/usr/bin/env bash

# Copyright 2020 The Kubernetes Authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

set -o errexit
```

Reply:
> In our infra-related scripts at k/k8s.io we are trying to be consistent with Google's shell styleguide so some of my comments will be related to this styleguide, but I'm not gonna block the PR because of this and leave you the decission if you would like to consider the suggestions.

The suggestions make sense, but I'd prefer to get the PR merged as-is first because I want the jobs in place before updating the individual repos.

Reply:
Sure, I don't have problems with the following up PR, but unfortunately I don't have approving powers so we'll have to wait for @spiffxp or others to approve it

Reply:
Script enhancements are in https://github.com/kubernetes/test-infra/pull/17808.